src/main/java/uk/gov/companieshouse/overseasentitiesapi/interceptor/UserAuthenticationInterceptor.java

```diff
@@ -52,18 +69,37 @@ public boolean preHandle(@NonNull HttpServletRequest request, @NonNull HttpServl
         final var tokenPermissions = getTokenPermissions(request)
                 .orElseThrow(() -> new IllegalStateException("UserAuthenticationInterceptor - TokenPermissions object not present in request"));
 
-        // Check the user has the company_incorporation=create permission
-        boolean hasCompanyIncorporationCreatePermission = tokenPermissions.hasPermission(Key.COMPANY_INCORPORATION, Value.CREATE);
+        boolean isRoeUpdateJourneyRequest = isRequestRoeUpdate(request);
 
         var authInfoMap = new HashMap<String, Object>();
         authInfoMap.put(TRANSACTION_ID_KEY, transactionId);
         authInfoMap.put("request_method", request.getMethod());
-        authInfoMap.put("has_company_incorporation_create_permission", hasCompanyIncorporationCreatePermission);
+        authInfoMap.put("is_roe_update_journey_request", isRoeUpdateJourneyRequest);
 
-        if (hasCompanyIncorporationCreatePermission) {
-            ApiLogger.debugContext(reqId, "UserAuthenticationInterceptor authorised with company_incorporation=create permission",
-                    authInfoMap);
-            return true;
+        if (isRoeUpdateJourneyRequest) {
+            boolean doCompanyNumbersMatch = doCompanyNumbersMatch(request, response, transactionId);
+            // Check the user has the company_oe_annual_update=create permission for ROE Update
+            boolean hasCompanyOeAnnualUpdateCreatePermission = tokenPermissions.hasPermission(Key.COMPANY_OE_ANNUAL_UPDATE, Value.CREATE);
```

Comment:
Maybe check token permission first - denial of service protection so that request has to have right scope before expensive tests are done.